File: backend/app/schemas/employee.py
import datetime as dt

from pydantic import BaseModel, ConfigDict, EmailStr


class EmployeeBase(BaseModel):
    name: str
    email: EmailStr
    role: str


class EmployeeCreate(EmployeeBase):
    pass


class EmployeeUpdate(BaseModel):
    name: str | None = None
    email: EmailStr | None = None
    role: str | None = None


class EmployeeRead(EmployeeBase):
    model_config = ConfigDict(from_attributes=True)

    id: int


class EmployeeAvailabilitySummary(BaseModel):
    as_of_date: dt.date
    allocation_percentage: int
    available_percentage: int
    status: str
    next_full_availability_date: dt.date | None = None


class EmployeeAvailabilityRead(EmployeeRead):
    availability: EmployeeAvailabilitySummary

File: frontend/src/types/employee.ts
export interface Employee {
  id: number;
  name: string;
  email: string;
  role: string;
}

export interface EmployeeAvailability {
  as_of_date: string;
  allocation_percentage: number;
  available_percentage: number;
  status: "free" | "occupied" | "partially_occupied";
  next_full_availability_date: string | null;
}

export interface EmployeeWithAvailability extends Employee {
  availability: EmployeeAvailability;
}
